update docs around legacy default OIDC subject (#12756)
pulumi/docs: content/blog/esc-projects-environment-tags-launch/index.md

  pulumi/cloud-pe/demo
 ```

> ### Breaking Changes

Reviewer: ```suggestion
### OIDC Subject Changes
```

Author: The changes here are both for OIDC subjects as well as environment context

Reviewer: Updating! Didn't want to call them "breaking changes" because it isn't.

> All existing environments should continue to work with the introduction of Projects, belonging now to a `default` project. However, when moving existing environments out of the default project you may notice some differences that we have outlined below to be proactive in communicating possible actions you'll need to migrate your environments.

Reviewer 2: Can we split the last sentence into 2 sentences? it's kind of run on

> All existing environments should continue to work with the introduction of Projects, belonging now to a `default` project. However, when moving existing environments out of the default project you may notice some differences that we have outlined below to be proactive in communicating possible actions you'll need to migrate your environments.

Reviewer: ```suggestion
With the introduction of [Projects](/blog/esc-projects-environment-tags-launch/), all existing environments will now be part of the `default` project and will continue to work without requiring any changes. However, when moving existing environments out of the default project you may notice some differences that we have outlined below.```

> * If your environment was interpolating the `context.currentEnvironment.name` or `context.rootEnvironment.name` from [Contextual information](/docs/esc/environments/#pulumi-contextual-information) this value will now include the project for any environments outside of the `default` project.

Reviewer 2: comma after `contextual-information),`

> * If your environment was interpolating the `context.currentEnvironment.name` or `context.rootEnvironment.name` from [Contextual information](/docs/esc/environments/#pulumi-contextual-information) this value will now include the project for any environments outside of the `default` project.

Reviewer: Can we add an example here?

Author: I link to the docs which I think should be enough

> * If you were using any of the [secrets providers with OIDC](/docs/esc/environments/#using-secrets-providers-and-oidc), the subject will be either `pulumi:environments:org:<organization name>:env:<project name>/<environment name>` if no `subjectAttributes` are specified or if the `currentEnvironment.name` subject attribute is specified it will now resolve to `currentEnvironment.name:<project name>/<environment name>`.

Reviewer: ```suggestion
* If you were using any of the [providers with OIDC](/docs/esc/environments/#using-secrets-providers-and-oidc), the subject will be either `pulumi:environments:org:<organization name>:env:<project name>/<environment name>` if no `subjectAttributes` are specified or if the `currentEnvironment.name` subject attribute is specified it will now resolve to `currentEnvironment.name:<project name>/<environment name>`.
```

Author: going to keep as secrets providers since that's how it's referenced in the concepts doc

> * If you were using any of the [secrets providers with OIDC](#using-secrets-providers-and-oidc) the subject will be either `pulumi:environments:org:<organization name>:env:<project name>/<environment name>` if no `subjectAttributes` are specified or if the `currentEnvironment.name` subject attribute is specified it will now resolve to `currentEnvironment.name:<project name>/<environment name>`. Updating this value will require manual intervention for the secrets provider in the new environment to work.

Reviewer 2: same `[secrets providers with OIDC](#using-secrets-providers-and-oidc),`

> * If you were using any of the [secrets providers with OIDC](#using-secrets-providers-and-oidc) the subject will be either `pulumi:environments:org:<organization name>:env:<project name>/<environment name>` if no `subjectAttributes` are specified or if the `currentEnvironment.name` subject attribute is specified it will now resolve to `currentEnvironment.name:<project name>/<environment name>`. Updating this value will require manual intervention for the secrets provider in the new environment to work.

Reviewer: ```suggestion
* If you were using any of [Pulumi ESC providers](/docs/esc/providers/) with OIDC, the subject will be either `pulumi:environments:org:<organization name>:env:<project name>/<environment name>` if no `subjectAttributes` are specified or if the `currentEnvironment.name` subject attribute is specified it will now resolve to `currentEnvironment.name:<project name>/<environment name>`.
```

Author: going to keep as secrets providers as that's how it's referenced on the concepts page
